tasks/libs/common/datadog_api.py

```diff
@@ -1,41 +1,37 @@
-def create_count(metric_name, timestamp, value, tags):
+from invoke.exceptions import Exit
+
+
+def create_metric(metric_type, metric_name, timestamp, value, tags, unit=None):
+    """
+    - metric_type: See types in the following documentation https://datadoghq.dev/datadog-api-client-python/datadog_api_client.v2.model.html#module-datadog_api_client.v2.model.metric_intake_type
+    """
+    from datadog_api_client.model_utils import unset
     from datadog_api_client.v2.model.metric_intake_type import MetricIntakeType
     from datadog_api_client.v2.model.metric_point import MetricPoint
     from datadog_api_client.v2.model.metric_series import MetricSeries
 
+    unit = unit or unset
+
     return MetricSeries(
         metric=metric_name,
-        # 1 is the count type
-        # https://datadoghq.dev/datadog-api-client-python/datadog_api_client.v2.model.html#module-datadog_api_client.v2.model.metric_intake_type
-        type=MetricIntakeType(1),
+        type=MetricIntakeType(metric_type),
         points=[
             MetricPoint(
                 timestamp=timestamp,
                 value=value,
             )
         ],
         tags=tags,
+        unit=unit,
     )
 
 
-def create_gauge(metric_name, timestamp, value, tags):
-    from datadog_api_client.v2.model.metric_intake_type import MetricIntakeType
-    from datadog_api_client.v2.model.metric_point import MetricPoint
-    from datadog_api_client.v2.model.metric_series import MetricSeries
+def create_count(metric_name, timestamp, value, tags, unit=None):
+    return create_metric(1, metric_name, timestamp, value, tags, unit)
 
-    return MetricSeries(
-        metric=metric_name,
-        # 3 is the gauge type
-        # https://datadoghq.dev/datadog-api-client-python/datadog_api_client.v2.model.html#module-datadog_api_client.v2.model.metric_intake_type
-        type=MetricIntakeType(3),
-        points=[
-            MetricPoint(
-                timestamp=timestamp,
-                value=value,
-            )
-        ],
-        tags=tags,
-    )
+
+def create_gauge(metric_name, timestamp, value, tags, unit=None):
+    return create_metric(3, metric_name, timestamp, value, tags, unit)
```

Comment:
Could we switch to the constant instead of having a magic number? https://github.com/DataDog/datadog-api-client-python/blob/fe2ba68837495e845f62442cedc1bdbbe939126f/src/datadog_api_client/v2/model/metric_intake_type.py#L41-L44